### DISCUSSION TITLE
react-hook-form and zod to validate form fields

### DISCUSSION THREAD

**Discussion by DanPresa** [colinhacks/zod#1517] [Q&A] [answered]
Hello there everyone.
I have a form and I'm also using Mui v5. I have an autocomplete but my problem is that I dont know how to validate it as a required. 
I already know how to control it in the front end but I want to custom an error message.
For instance.

```
const zodSchema = z.object({
  status: z.object({
     label: z.string().min(1),
     value: z.string().min(1)
  })
}).required()
```

**Comment by MerlinMason** [comment]
I think the following should work?
```ts
z.string().min(1, { message: "example" })
```

**Reply by DanPresa** [reply]
Unfortunately It doesn't work

**Reply by MerlinMason** [reply]
ohh also just noticed your usage of `.required()`... Zod follows the typescript methodology of everything being required by default, unless you call `.optional()`.

**Reply by DanPresa** [reply, marked as answer]
@MerlinMason Thanks for the reply. I decided using yup instead zod. It was a better option and I can validate autocomplete

**Reply by neumadhu** [reply]
I am also facing the same issue with MUI + react hook form + zod.  Please let me know if we have any solution for this.
